pysurge/pysurge.py

```diff
@@ -183,19 +183,25 @@ def _thread_pool_submitter(self):
         """Handles the continuous loop to submit tests to the thread pool executor."""
         start_time = time.time()
 
-        rate_of_fire = 1.0 / self.rate
         max_duration = self.test_instance.max_duration
-        workers = rate_of_fire * max_duration
+        # Based on Little's Law
+        # Max number of concurrent tests = (num tests fired/sec) * (max duration in sec of test)
+        workers = self.rate * max_duration
 
         executor = ThreadPoolExecutor(max_workers=workers)
-        last_submit_time = 0.0
+
+        rate_of_fire = 1.0 / self.rate  # 1 test fired every 'rate_of_fire' seconds
 
         while not self._stop.is_set():
-            if time.time() - last_submit_time >= rate_of_fire:
-                last_submit_time = time.time()
+            try:
                 executor.submit(self._test_runner)
-            # Sleep for short durations so we can quickly catch the _stop flag being set
-            time.sleep(0.1)
+            except RuntimeError as err:
+                if str(err) == "can't start new thread":
```

Comment:
Perhaps we should change this to be:
```suggestion
                if "can't start new thread" in str(err):
```
in case they change the exception down the road to include more text.